lib/kafka_rest_client.js

```diff
@@ -272,9 +272,9 @@ KafkaRestClient.prototype.tryMakeRequest = function tryMakeRequest(produceMessag
             self.statsd.timing(metricPrefix + '.latency', time);
         }
 
-        if (err) {
+        if (err || (res && res.statusCode >= 500 && res.statusCode < 600)) {
```

Comment:
1. If err is not null, it will not check res; if err is null, res won't be null;
2. If it is 404, err will be null.